doc/develop/api/api_conventions.rst

```diff
@@ -0,0 +1,64 @@
+.. _api_conventions.rst:
+
+API Conventions
+###############
+
+Zephyr provides a few programming interfaces that users can use for creating their applications.
+Zephyr's API space includes the following areas: - C programming interface (function prototypes,
+structures, and macros), provided in C header files - Configuration system (Kconfig) - Device driver
+model (Devicetree)
+
+Other areas, such as the build system (CMake), may also be seen as Zephyr's API. However, any area
+that is not explicitly listed herein is not considered Zephyr's API as of today.
+
+API classification
+******************
+The files that define the Zephyr APIs may contain symbols intended for different usage. The intended
+API usage is designated by the API class. Zephyr APIs are classified as:
+
+- Private
+    These APIs are intended for use within boundary of a :term:`software component`. Private APIs
+    defined in Zephyr tree are not subject to :ref:`api_lifecycle`. Therefore, can be changed or
+    removed at any time. Changes to the private API may not be documented.
+
+- Internal
```

Comment:
From Process WG: Suggest adding language indicating that selection of APIs into public/internal classes is done on a case by case basis.

Reply:
Also add a sentence indicated that internal APIs can get promoted to public if there is justification (user creates an RFC/PR to propose the promotion.